numba/_monitoring.c

```diff
@@ -0,0 +1,360 @@
+/*
+This is a copy of the sys monitoring logic in _dispatcher.cpp for Python 3.12.
+This is necessary to workaround issue with g++ complaining about C++ template
+inside "internal/pycore_mimalloc.h" from #include "internal/pycore_interp.h" 
+*/
```

Comment:
I might be able to remove this workaround now, given https://github.com/python/cpython/pull/124273#issuecomment-2369244566
Having this C file compiling against C++ files (in dispatcher module) is also a mess. I have to drop `-Werror`